### DISCUSSION TITLE
Custom CSS

### DISCUSSION THREAD

**Discussion by larmarange** [retypeapp/retype#81] [Q&A]
Sorry but I didn't found if there is a way to include a custom CSS file in all built pages.

**Comment by geoffreymcgill** [comment]
At this time, there is no way built in way within Retype to include a custom CSS file. I'm not too sure we'll be including this feature either, although we will be including options to set a custom color palette soon. Work is being done right now on the color palette options.

**Reply by alisman** [reply]
In my opinion this is an essential feature.  There is just no way markdown can accommodate all use cases.   It's very nice that retype allows for html anywhere, but it would be great to be able to style it globally.

**Reply by geoffreymcgill** [reply]
Hi @alisman As of Retype `v2`, you can add a custom global `.css` file using the `_includes/head.html` include, see comment at https://github.com/retypeapp/retype/discussions/159#discussioncomment-2472860

Hope this helps.

**Comment by geoffreymcgill** [comment]
Out of interest, what CSS would you like to include or modify?

**Reply by larmarange** [reply]
Just to provide some context, we are a group of scientists writing an evolving open access book summarising current scientific knowledge in our field for a large audience (including civil society and policymakers). We are currently comparing several possible technical solutions for writing and rendering the book.

The idea would be to have some custom divs. For example:

- a list of bibliographic references would be added at the end of each chapter but we would like to format them differently (smaller font sizes in particular)
- we want to embed some multimedia content like  YouTube videos. However, if the page is printed, we would like to display instead a QRCode and a link to access the content. In such situation, it would be great to have a "noprint" and a "print-only" classes.

If I add in my markdwon file:

```md
::: custom_class

content of the div

:::
```

It will render

```html
<div class="custom_class">
content of the div
</div>
```

But where defining the CSS of `custom_class`?

**Reply by geoffreymcgill** [reply]
Thanks for the update. We are discussing some options and will follow-up soon with a plan. 

There will be a supported way within Retype to get this working easily. We just need to work through some ideas and figure out a super clean way to configure.

**Reply by DannyBen** [reply]
> ::: custom_class

This is a nice idea I think.

Personally, I love the opinionated styling that Retype chose, but adding "your own touch" to markdown files was always a little bit missing.

> Work is being done right now on the color palette options.

On a somewhat-related note - if I had to choose one thing I would love to be able to change in the generated Retype files, is the fact that even in the light theme, the code snippets are on a dark background. I wish to have an option in the **retype.~json~yml** 😉 file to state that I want code snippets to go "hand in hand" with the selected theme.

**Reply by geoffreymcgill** [reply]
Hi @larmarange Not sure if you're still interested, but the container syntax you proposed is now supported within Retype `v2`. 

The following Markdown element:
```md
::: custom_class

content of the div

:::
```
Creates the following HTML:
```html
<div class="custom_class">
    <p>content of the div</p>
</div>
```
You can now also add inline css. For example:
```md
<style>
    .custom_class {
        background-color: red;
    }
</style>
::: custom_class
content of the div
:::
```
Will render as:
<img width="335" alt="Screen Shot 2022-03-30 at 2 53 08 PM" src="https://user-images.githubusercontent.com/62210/160928984-38682596-a7c2-4ae7-9174-7a9de23d236d.png">

You can now also include a global `.css` by adding a `_includes/head.html` file. See comments at https://github.com/retypeapp/retype/discussions/282#discussioncomment-2343005

I'm working on the `_includes/head.html` documentation today, so there should be more details soon, but it's pretty straightforward functionality. Put any raw HTML in the `_includes/head.html` and it will added into the `<head>` of all pages in your website.

Hope this helps.

**Reply by geoffreymcgill** [reply]
@larmarange New Container component docs have been published to https://retype.com/components/container/.

Also includes a couple samples on applying inline `<style>` or linking to a separate `.css` using `_includes/head.html`.

Hope this helps.

**Reply by larmarange** [reply]
Thanks. That's perfect

**Reply by benkoshy** [reply]
This is handy. What would be even more ideal - for my particular use case - would be the ability to add [tailwind utility classes](https://github.com/retypeapp/retype/discussions/221) and have everything build quickly and just work like magic:

> ::: hover:bg-red-900
> content of the div
> :::
> ```

I prefer strongly not to use a separate stylesheets for most things. I understand making retype do this is noteasy, possible, or even worth it, for a sub-set of users (i.e. just me). i'm just throwing my two cents in.

**Comment by alisman** [comment]
Perfect!!! Works like a charm.

**Comment by AlexKatyukov** [comment]
Hi!

I want to have an ability to change fonts. Unfortunately, Poppins not working with cyrillic so it used next font, but punctuation marks are from Poppins.

![image](https://user-images.githubusercontent.com/18090165/176857108-77876d63-b1f7-4498-8d5f-b5a0fa4f330d.png)

As you can see, `()/` are weird. The first one have standart font-family, the second — the same but without Poppins ¯\\\_(ツ)_/¯

**Comment by zatmonkey** [comment]
I'd like to be able to align the style of retype generated documentation site to that of my main app

**Comment by dleunig** [comment]
Are there any updated regarding color palettes? Need to update the colors with CD.

**Comment by MrBns** [comment]
I would like to change the default font of retype. is there any of do that ?

**Comment by gosre** [comment]
Is there any update to color palettes?

**Comment by gatodelcaribe** [comment]
On the new version we got Themes to changue colors. You can check it here: https://retype.com/configuration/theme-variables/
